FileShareService.DesktopClient.Core/Jobs/SetExpiryDateJob.cs

```diff
@@ -1,27 +1,53 @@
-﻿
-using Newtonsoft.Json.Linq;
+﻿using Newtonsoft.Json.Linq;
+using System;
 using System.Collections.Generic;
 
 namespace UKHO.FileShareService.DesktopClient.Core.Jobs
 {
     public class SetExpiryDateJob : IJob
     {
         public const string JOB_ACTION = "setExpiryDate";
-        public string DisplayName { get; set; }
-        public string Action { get; set; }
+        public string DisplayName { get; set; } = string.Empty;
+        public string Action { get; set; } = string.Empty;
         public SetExpiryDateJobParams ActionParams { get; set; } = new SetExpiryDateJobParams();
         public List<string> ErrorMessages { get; private set; } = new List<string>();
 
+        // To hold whether expiry date is specified in config or not.
+        public bool IsExpiryDateKeyExist { get; private set; }
+
         public void Validate(JToken jsonToken)
         {
-            //This validation will be implemented when PBI SetExpiryDateJob is picked-up.
+            #region Predeserialize validations
+            JToken? expiryDateToken = jsonToken.SelectToken("actionParams.expiryDate");
+
+            //Set value if key exists
+            IsExpiryDateKeyExist = expiryDateToken != null;
```

Comment:
Rename variable from `IsExpiryDateKeyExist` to `ExpiryDateKeyExists` please

Reply:
Done rename to ExpiryDateKeyExists as suggested.